presto-main/src/main/resources/webapp/src/components/ResourceGroupView.jsx

```diff
@@ -0,0 +1,349 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { addToHistory, formatDataSizeBytes, parseDataSize, truncateString } from "../utils";
+import { QueryListItem } from "./QueryList";
+
+const SPARKLINE_PROPERTIES = Object.freeze({
+    disableHiddenCheck: true,
+    fillColor: '#3F4552',
+    height: '75px',
+    lineColor: '#747F96',
+    spotColor: '#1EDCFF',
+    tooltipClassname: 'sparkline-tooltip',
+    width: '100%',
+});
+
+// One information entry of the resource group
+function InfoItem({name, value}) {
+    return (
+        <tr>
+            <td className="info-title">
+                {name}
+            </td>
+            <td className="info-text wrap-text">
+                {value}
+            </td>
+        </tr>
+    );
+}
+
+// One timeline chart of the resource group
+function TimelineItem({name, chartId, value}) {
+    return (
+        <>
+        <tr>
+            <td className="info-title">
+                {name}
+            </td>
+            <td rowSpan="2">
+                <div className="query-stats-sparkline-container">
+                    <span className="sparkline" id={chartId}><div className="loader">Loading ...</div></span>
+                </div>
+            </td>
+        </tr>
+        <tr className="tr-noborder">
+            <td className="info-sparkline-text">
+                {value}
+            </td>
+        </tr>
+        </>
+    );
+}
+
+// Display the root group list if no group is specified or general information if any error.
+function NoGroupIdWidget({groupId, error, groups}) {
+    let docView = (<h3 className="text-center">{ groupId ? 'Retrieving resource group information...' : 'Detecting Resource Groups settings...'}</h3>);
+
+    if (groups.length > 0) {
+        docView = (
+            <>
+            <div className="col-xs-4 col-xs-offset-4">
+                <h4 className="text-center">Available resource groups:</h4>
+                <div class="list-group">
```

Comment:
Please check whether the `list-group` CSS class is needed here and consider replacing `class` [with `className`.](https://legacy.reactjs.org/docs/dom-elements.html#classname)


```suggestion
                <div className="list-group">
```